### DISCUSSION TITLE
photosensitivity/epilepsy

### DISCUSSION THREAD

**Discussion by geextahslex** [mpv-player/mpv#14279] [General]
Hi, some videos have flashing lights, in clubs scenes or muzzle flashes in shootouts, which can be hard on the eyes. I would like to know if there is any filter/shader/function etc. to deal with it. 

Thank you :)

**Comment by kasper93** [comment]
Maybe @junguler has some tips for this. I recall they were using frei0r for compressing overly bright areas https://github.com/mpv-player/mpv/discussions/14211 Generally it is possible with filters, depends what effect is needed exactly.

**Comment by richardpl** [comment]
Trying https://ffmpeg.org//ffmpeg-filters.html#photosensitivity ?

**Reply by geextahslex** [reply]
Thank you for your answer. How can I run this in mpv?

**Reply by bungiefan** [reply]
> Thank you for your answer. How can I run this in mpv?

This is also my concern. I've been directed to this as a fix for old anime that strobes heavily, as my wife is epileptic. El Hazard and Tenchi Muyo and some other stuff with lightning/electricity effects flashes very rapidly and sets her off. Trying to use it in VLC/MPC on Windows or Linux just seems to not be effective. I'm finding a lot of shows from the 80s to the 2000s that just aren't watchable with her.

**Reply by junguler** [reply]
> > Thank you for your answer. How can I run this in mpv?
> 
> This is also my concern. I've been directed to this as a fix for old anime that strobes heavily, as my wife is epileptic. El Hazard and Tenchi Muyo and some other stuff with lightning/electricity effects flashes very rapidly and sets her off. Trying to use it in VLC/MPC on Windows or Linux just seems to not be effective. I'm finding a lot of shows from the 80s to the 2000s that just aren't watchable with her.

hello, since the last time i've posted here i have made a "eye-saver" .cube lut that you can use with mpv and it completely removes white colors from the video and turns them to black, it might look weird but it does the job

https://github.com/junguler/win_confs/blob/main/mpv/eye_saver_16.cube

save the [raw](https://github.com/junguler/win_confs/raw/refs/heads/main/mpv/eye_saver_16.cube) file to your hard drive somewhere and load it via a keyboard shortcut like this (put this in your input.conf)

```
f1 vf toggle lut3d="C:\bin\eye_saver_16.cube" ; show-text "Eye Saver"
```

this assumes your lut is in C:/bin and is named eye_saver_16.cube , activate it by pressing f1 in your keyboard

this is the harshest level of this lut tho, it can be made to be less damaging to the video but i didn't want to take any risks with my eyesight

here is a before and after example on an anime

![before](https://github.com/user-attachments/assets/88a8cc2b-5394-41c4-ad2f-ff3605376a67)
![after](https://github.com/user-attachments/assets/397dc9e1-517a-4bce-b9a0-05fc4297fb76)

if you kind of like it but want the effect to be less intrusive you can feed the same .cube lut to chatgpt or some other chatbot and ask it to undo it about 50% or so

i hope this helps as i also struggled a lot with white flashes when watching movies and anime in the past, same method can also be applied to games using SweetFX

**Comment by junguler** [comment]
yes, frei0r has a filter called colortap, it helps a lot with bright flashes of white but it looks kind of weird, you can find a mpv build that has frei0r compiled and test it out yourself and see how you like/dislike it

you can use this extension for live testing filters btw, makes it really easy to quickly test them https://github.com/hdb/mpv-live-filters
